src/trie.rs

```diff
@@ -88,15 +99,18 @@ impl<T: Clone> Map<String, T> for PatriciaTrie<T> {
         match *self {
             Tip => panic!("lookup on empty tree."),
             Node { ref key, ref value, ref children } => {
-                if k == *key {
-                    match *value {
-                        Some(ref v) => v.clone(),
-                        None => panic!("element does not exist"),
-                    }
-                } else if k.starts_with(key) {
-                    match children.get(&k.char_at(key.len())) {
```

Comment:
I think we can just change this line to `match children.get(&k.chars().nth(key.len()).unwrap()) {` and leave everything else as is.